Various improvements

steveh-bipsync opened the pull request:
- Adds the ability to limit the tokens to a certain number, eg. 1
  - If the token limit has been reached replace the previous token added
- Add a method to replace a token
  - Rather than having to call removeToken then addToken
- Method to get token data for a given token element
- New feature to optionally show suggestions (with a limit) on focus
- Fix issue where disableFocusOnTokenClick was not being checked in all cases
- Adds a method to toggle readonly state of the token input

stevehalford committed:
Add a token limit

If the token limit has been reached replace the previous token added

stevehalford committed:
Add a tokeninput method to replace a token

Rather than having to call removeToken then addToken

stevehalford committed:
Tokeninput method to get token data for a given token element

stevehalford committed:
Tokeninput feature to show some suggestions on focus

stevehalford committed:
Another check for disableFocusOnTokenClick

stevehalford committed:
Adds a method to toggle readonly state of the token input

paulatbipsync commented on tokeninput.js:
```diff
@@ -982,10 +997,28 @@
+
+            this.removeToken(this.tokens[this.tokens.length - 1], {
+                silent: true
+            });
```
What if the number of tokens is more than 1 higher than the limit?

steveh-bipsync replied:
Will refactor

steveh-bipsync replied:
Have done this

paulatbipsync reviewed (changes requested):
In this library we use the word "completions", so the change should probably follow that convention.

stevehalford committed:
Update vocabulary: Suggestions > Completions

stevehalford committed:
Refactor replacing the last token when at the limit

On adding a new token, if we have a token limit and we are over the limit then remove tokens until we are one under the limit to make room for the new token

steveh-bipsync commented:
> In this library we use the word "completions", so the change should probably follow that convention.

Have renamed suggestions > completions

stevehalford committed:
Remove reference to container click handler when setting readonly
